libbeat/outputs/redis/config.go

```diff
@@ -39,3 +41,17 @@ var (
 		DbTopology:       1,
 	}
 )
+
+func (c *redisConfig) Validate() error {
+	switch c.DataType {
+	case "", "list", "channel":
+	default:
+		return fmt.Errorf("redis data type %v not supported", c.DataType)
+	}
+
+	if c.Index == "" {
+		return errors.New("index required")
```

Comment:
I wonder if we need more context in the error message like "redis index required"?